recipe-server/normandy/recipes/tests/test_models.py

```diff
@@ -50,6 +52,40 @@ def test_recipes_used_by_empty(self):
         RecipeFactory.create_batch(2, action=action, enabled=False)
         assert list(action.recipes_used_by) == []
 
+    def test_update_signature(self, mocker, mock_logger):
+        # Mock the Autographer
+        mock_autograph = mocker.patch('normandy.recipes.models.Autographer')
+        mock_autograph.return_value.sign_data.return_value = [
+            {'signature': 'fake signature'},
+        ]
+
+        action = ActionFactory(signed=False)
+        action.update_signature()
+        mock_logger.info.assert_called_with(
+            Whatever.contains(action.name),
+            extra={'code': INFO_REQUESTING_ACTION_SIGNATURES, 'action_names': [action.name]}
+        )
+
+        action.save()
+        assert action.signature is not None
+        assert action.signature.signature == 'fake signature'
+
+    def test_canonical_json(self):
+        action = ActionFactory(name='test-action', implementation='console.log(true)')
+        # Yes, this is ugly, but it needs to compare an exact byte
+        # sequence, since this is used for hashing and signing
+        expected = (
+            f'{{'
```

Comment:
I don't see any formatting happening, we can probably just make these all normal strings.